app/backend/src/clj/teet/file/file_commands.clj

```diff
@@ -71,6 +71,18 @@
                 ;; Take part after last underscore to skip over invalid metadata
                 (last (str/split n #"_")))))))
 
+(defcommand :file/upload-complete
+  {:doc "Mark file upload as complete"
+   :context {:keys [conn user db]}
+   :payload {id :db/id}
+   :pre [(file-db/own-file? db user id)]
+   ;; No need to check extra authorization again, as we check pre condition that this
+   ;; is the user's own uploaded file
+   :project-id nil
+   :authorization {}
+   :transact [{:db/id id
+               :file/upload-complete? true}]})
```

Comment:
So what happens if the upload actually completes but the client doesn't call this?

Reply:
Due to e.g. page reload.

Reply:
the file is forever incomplete and will not show on the listing

Reply:
but that can be queried for by searching for files that are created long ago but don't have upload-complete yet